app/assets/javascripts/application.js

```diff
@@ -27,20 +27,36 @@
 //= require components/editor
 
 $(function() {
-    autosize($('.field textarea'));
-    autosize.update($('.field textarea'));
+	autosize($('.field textarea'));
+	autosize.update($('.field textarea'));
 });
 
 $(function() {
-    if($('[data-choices]').length > 0) {
-        new Choices('[data-choices]', {shouldSort: false});
-    }
+	if($('[data-choices]').length > 0) {
+		new Choices('[data-choices]', {shouldSort: false});
+	}
 });
 
 $(function() {
-    $('.range').range();
-
-    $('.trigger-conditions').on('cocoon:before-insert', function(e, item) {
-        $(item).find('.range').range();
-    });
-});
\ No newline at end of file
+	$('.range').range();
+
+	$('.trigger-conditions').on('cocoon:before-insert', function(e, item) {
+		$(item).find('.range').range();
+	});
+
+	$('.trigger-conditions').on('cocoon:after-insert', function(e, item) {
+		$('select.choose_sensor').on('change', function() {
+			var options = $(this).find('option:selected')[0].dataset
+
+			$('.range').data('range').reinit({
+				min: options.rangeMin,
+				max: options.rangeMax,
+				step: options.rangeStep,
+				valueFormatter: function(value) {
+					return value +  options.rangeUnit;
+				}
+			});
+
+		});
+	});
+});
```

Comment:
You can merge those two event handlers. It doesn’t matter wether you’re setting the event handlers on the select before or after adding it to the DOM.